sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -502,196 +506,178 @@ func (s *NetworkFlowManagerTestSuite) TestEnrichProcessListening() {
-			connections: []*HostnameAndConnections{
-				createHostnameConnections(hostname).withEndpointPair(createEndpointPair(timestamp.Now()).containerID(fmt.Sprintf("%s-1", containerID))),
-				createHostnameConnections(hostname).withEndpointPair(createEndpointPair(timestamp.Now()).containerID(fmt.Sprintf("%s-2", containerID))),
-			},
-		},
-		{
-			testName: "In online mode we should enrich and send the previously received endpoints",
-			notify:   common.SensorComponentEventCentralReachable,
-			expectEntityLookupContainer: func() {
-				gomock.InOrder(
-					mockEntity.EXPECT().LookupByContainerID(gomock.Any()).Times(1).DoAndReturn(func(_ any) (clusterentities.ContainerMetadata, bool) {
-						return clusterentities.ContainerMetadata{DeploymentID: fmt.Sprintf("%s-1", srcID)}, true
-					}),
-					mockEntity.EXPECT().LookupByContainerID(gomock.Any()).Times(1).DoAndReturn(func(_ any) (clusterentities.ContainerMetadata, bool) {
-						return clusterentities.ContainerMetadata{DeploymentID: fmt.Sprintf("%s-2", srcID)}, true
-					}),
-				)
-			},
-			expectedSensorMessage: createExpectedSensorMessageWithEndpoints(
-				fmt.Sprintf("%s-1", srcID),
-				fmt.Sprintf("%s-2", srcID),
-			),
+			Entity:         networkgraph.Entity{ID: dstID},
+			ContainerPorts: []uint16{80},
 		},
+	})()
+}
+
+func (b *sendNetflowsSuite) expectDetections(n int) {
+	expectDetectorHelper(b.mockDetector, n)()
+}
+
+func (b *sendNetflowsSuite) TestUpdateConnectionGeneratesNetflow() {
+	b.expectLookups(1)
+	b.expectDetections(1)
+
+	b.updateConn(createConnectionPair())
+	b.thenTickerTicks()
+	b.assertOneUpdatedConnection()
+}
+
+func (b *sendNetflowsSuite) TestUnchangedConnection() {
+	b.expectLookups(2)
+	b.expectDetections(1)
+
+	pair := createConnectionPair()
+	pair.status.lastSeen = timestamp.InfiniteFuture
+	b.updateConn(pair)
+	b.thenTickerTicks()
+	b.assertOneUpdatedConnection()
+
+	// There should be no second update, the connection did not change
+	b.thenTickerTicks()
+	mustNotRead(b.T(), b.m.sensorUpdates)
+}
+
+func (b *sendNetflowsSuite) TestSendTwoUpdatesOnConnectionChanged() {
+	b.expectLookups(2)
+	b.expectDetections(2)
+
+	pair := createConnectionPair()
+	oneHourAgo := timestamp.NowMinus(time.Hour)
+	pair.status.lastSeen = timestamp.FromProtobuf(oneHourAgo)
+	b.updateConn(pair)
+	b.thenTickerTicks()
+	b.assertOneUpdatedConnection()
+
+	pair.status.lastSeen = timestamp.Now()
+	b.updateConn(pair)
+	b.thenTickerTicks()
+	b.assertOneUpdatedConnection()
+}
+
+func (b *sendNetflowsSuite) TestUpdatesGetBufferedWhenUnread() {
+	b.expectLookups(4)
+	b.expectDetections(4)
+
+	// four times without reading
+	for i := 0; i < 4; i++ {
+		ts := timestamp.NowMinus(time.Duration(4-i) * time.Hour)
+		pair := createConnectionPair()
+		pair.status.lastSeen = timestamp.FromProtobuf(ts)
+		b.updateConn(pair)
+		b.thenTickerTicks()
+		time.Sleep(100 * time.Millisecond) // Immediately ticking without waiting causes unexpected behavior
```

Comment:
Note to reviewer: I've used 100ms here because that's the time that was used for the tests as ticker value.